web-app/src/lib/services/job/service.ts

```diff
@@ -120,33 +120,40 @@ export async function startJob(input: StartJobInputSchemaType): Promise<Job> {
 }
 
 export async function syncJob(job: Job) {
-  await prisma.$transaction(async (tx) => {
-    job = await syncRegistryStatus(job, tx);
-    job = await syncAgentJobStatus(job, tx);
-
-    const jobStatus = computeJobStatus(job);
-    switch (jobStatus) {
-      case JobStatus.PAYMENT_FAILED:
-      case JobStatus.REFUND_RESOLVED:
-        await refundJob(job.id, tx);
-        break;
-      default:
-        break;
-    }
-    switch (job.onChainStatus) {
-      case OnChainJobStatus.RESULT_SUBMITTED:
-        const completedAt = job.completedAt ?? new Date();
-        if (
-          job.agentJobStatus !== AgentJobStatus.COMPLETED &&
-          new Date().getTime() - completedAt.getTime() > 10 * 60 * 1000 // 10 minutes
-        ) {
-          await postPaymentClientRequestRefund(job);
-        }
-        break;
-      default:
-        break;
-    }
-  });
+  await prisma.$transaction(
+    async (tx) => {
+      job = await syncRegistryStatus(job, tx);
+      job = await syncAgentJobStatus(job, tx);
+
+      const jobStatus = computeJobStatus(job);
+      switch (jobStatus) {
+        case JobStatus.PAYMENT_FAILED:
+        case JobStatus.REFUND_RESOLVED:
+          await refundJob(job.id, tx);
+          break;
+        default:
+          break;
+      }
+
+      switch (job.onChainStatus) {
+        case OnChainJobStatus.RESULT_SUBMITTED:
+          const completedAt = job.completedAt ?? new Date();
+          if (
+            job.agentJobStatus !== AgentJobStatus.COMPLETED &&
+            new Date().getTime() - completedAt.getTime() > 10 * 60 * 1000 // 10 minutes
+          ) {
+            await postPaymentClientRequestRefund(job);
+          }
+          break;
+        default:
+          break;
+      }
+    },
+    {
+      maxWait: 5000, // default: 2000
+      timeout: 20000, // default: 5000
```

Comment:
Having this long of a lock is a major db overhead, maybe you can extract some stuff before the tx?

Reply:
Can you please give more context or an example? It seems like that you have something in mind if you ask like that.

Reply:
@schaier-io I extracted the requests to the registry and agent in cc26015a448b17514a11cef5a8112ff2f89bf9fb